nixos/modules/services/web-servers/webhook.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  webhook = pkgs.webhook;
+  user = cfg.user;
+  settingsJSON = pkgs.writeText "webhookSettings" (builtins.toJSON cfg.settings);
+
+in {
+  options.services.webhook = {
+
+    enable = mkOption {
+      default = false;
+      description = "Enable webhook service.";
+      type = types.bool;
+    };
+
+    user = mkOption {
+      default = "webhook";
+      description = "User webhook service runs as.";
+      type = types.str;
+    };
+
+    group = mkOption {
+      default = null;
+      description = "Group webhook service runs as.";
+      type = with types; nullOr str;
+    };
+
+    settings = mkOption {
+      default = null;
+      description = "Webhook service settings.";
+      type = with types; listOf attrs;
+      example = literalExpression ''
+        [{
+          id = "redeploy-webhook";
+          execute-command = "/var/scripts/redeploy.sh";
+          command-working-directory = "/var/webhook";
+        }]
+      '';
+    };
+
+    path = mkOption {
+      default = [];
+      description = "Which packages the service can access";
+      example = literalExpression ''
+        with pkgs; [ nixFlakes ]
+      '';
+      type = with types; listOf package;
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.webhook = {
+      description = "Webhook service";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = cfg.path;
+      serviceConfig = {
+        Type = "simple";
+        User = user;
+        WorkingDirectory = "${webhook}";
+        ExecStart = ''
+          ${webhook}/bin/webhook -hooks ${settingsJSON} -verbose
+        '';
+      };
+    };
+
+    users.users = mkIf (user == "webhook") {
+      webhook.isSystemUser = true;
+    };
```

Comment:
users and groups should only be configured if not explicitly set. other modules default them both to `null` and create the user/group only if it remains `null` with the default name for the module.